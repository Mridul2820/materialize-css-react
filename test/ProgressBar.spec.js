import React from 'react';
import { render } from '@testing-library/react';
import ProgressBar from '../src/ProgressBar';

describe('<ProgressBar />', () => {
  test('renders', () => {
    const { container } = render(<ProgressBar />);
    expect(container).toMatchSnapshot();
  });

  test('adds progress style', () => {
    const { container } = render(<ProgressBar progress={32} />);
    expect(container).toMatchSnapshot();
  });

  test('handles determinate', () => {
    const { container } = render(<ProgressBar progress={32} />);
    expect(container.querySelector('.determinate')).toBeTruthy();
  });

  test('handles indeterminate', () => {
    const { container } = render(<ProgressBar />);
    expect(container.querySelector('.indeterminate')).toBeTruthy();
  });
});
