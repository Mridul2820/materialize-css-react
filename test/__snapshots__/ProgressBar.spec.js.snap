// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBar /> adds progress style 1`] = `
<div>
  <div
    class="progress lighten-5"
  >
    <div
      class="determinate"
      style="width: 32%;"
    />
  </div>
</div>
`;

exports[`<ProgressBar /> renders 1`] = `
<div>
  <div
    class="progress lighten-5"
  >
    <div
      class="indeterminate"
    />
  </div>
</div>
`;
